# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/patches/with_update/fix_incorrect_int_use_in_Kernel32BaseVersion.patch
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/patches/with_update/fix_incorrect_int_use_in_Kernel32BaseVersion.patch
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/security/sandbox/chromium-shim/patches/with_update/fix_incorrect_int_use_in_Kernel32BaseVersion.patch	Wed Jun 12 11:10:48 2019 +0100
@@ -0,0 +1,40 @@
+# HG changeset patch
+# User Bob Owen <bobowencode@gmail.com>
+# Date 1549903491 0
+#      Mon Feb 11 16:44:51 2019 +0000
+# Node ID 3bb3e5a6275db53530d50549b7b329cc1248d450
+# Parent  fc9d5e40028058abeb06f8bdc2f75fc868fbe14b
+
+Fix incorrect int use in Kernel32BaseVersion in windows_version.cc. r=aklotz
+
+diff --git a/security/sandbox/chromium/base/win/windows_version.cc b/security/sandbox/chromium/base/win/windows_version.cc
+--- a/security/sandbox/chromium/base/win/windows_version.cc
++++ b/security/sandbox/chromium/base/win/windows_version.cc
+@@ -236,23 +236,23 @@ Version OSInfo::Kernel32Version() const 
+ // kernel32 will still be the "real" version.
+ base::Version OSInfo::Kernel32BaseVersion() const {
+   static const base::NoDestructor<base::Version> version([] {
+     std::unique_ptr<FileVersionInfoWin> file_version_info(
+         static_cast<FileVersionInfoWin*>(
+             FileVersionInfoWin::CreateFileVersionInfo(
+                 base::FilePath(FILE_PATH_LITERAL("kernel32.dll")))));
+     DCHECK(file_version_info);
+-    const int major =
++    const uint32_t major =
+         HIWORD(file_version_info->fixed_file_info()->dwFileVersionMS);
+-    const int minor =
++    const uint32_t minor =
+         LOWORD(file_version_info->fixed_file_info()->dwFileVersionMS);
+-    const int build =
++    const uint32_t build =
+         HIWORD(file_version_info->fixed_file_info()->dwFileVersionLS);
+-    const int patch =
++    const uint32_t patch =
+         LOWORD(file_version_info->fixed_file_info()->dwFileVersionLS);
+     return base::Version(std::vector<uint32_t>{major, minor, build, patch});
+   }());
+   return *version;
+ }
+ 
+ std::string OSInfo::processor_model_name() {
+   if (processor_model_name_.empty()) {